Feature/register frontend

jasonevangelista opened the pull request:
- register modal frontend
- input validation
- unit tests for modal inputs (except for e-mail input since may validation check for existing email sa db, di kaya sa unit test)

Let me know if may suggestions kayo sa design and colors! (I made the main color blue lang muna since it might be weird if red kasi yung red color sa modal nag iindicate if wrong input so baka magka confusion)

Screenshots of modal:

- Initial View of Modal

![Screen Shot 2020-12-27 at 2 13 29 PM](https://user-images.githubusercontent.com/50069439/103165084-70ad8100-484e-11eb-8ae5-9dece2e05d8e.png)

- Sample Error Displays

![Screen Shot 2020-12-27 at 2 20 58 PM](https://user-images.githubusercontent.com/50069439/103165114-c2560b80-484e-11eb-8efa-8e88dc1cfc69.png)

- Success Displays

![Screen Shot 2020-12-27 at 2 14 01 PM](https://user-images.githubusercontent.com/50069439/103165119-c6822900-484e-11eb-8ec5-f01b6b02a50d.png)

- Password Info Display

![Screen Shot 2020-12-27 at 2 14 14 PM](https://user-images.githubusercontent.com/50069439/103165125-d26deb00-484e-11eb-92da-d25c1763b2db.png)

jasonevangelista committed:
added initial register modal UI

jasonevangelista committed:
added border-radius override for modal

jasonevangelista committed:
Merge branch 'develop' of https://github.com/jasonevangelista/stsweng-zarap into feature/register-frontend

jasonevangelista committed:
adjusted css to limit overrides to search bar only

jasonevangelista committed:
adjust register modal width and added initial unit tests

jasonevangelista committed:
call local onFinish method before onFinish prop method

jasonevangelista committed:
added complete unit tests for register modal inputs

jasonevangelista committed:
adjusted overrides

jasonevangelista committed:
added temp colors for register button

jasonevangelista committed:
added email validation

jasonevangelista committed:
updated unit tests for register modal

jasonevangelista committed:
added tooltip display for password requirements

jasonevangelista committed:
adjusted btn override

jasonevangelista committed:
added fix

jasonevangelista committed:
modified input field checker test

jasonevangelista committed:
adjusted tooltip name to match input field test

jasonevangelista committed:
spacing adjusted by linter

jasonevangelista committed:
fixed linting problems for promises and register button in header

NeilMatthewLua commented:
Hi @jasonevangelista! Thank you for your continuous effort on this project! 

While testing on your branch, I've noticed that whenever I click on the Sign-Up button, the scroll bar on the side vanishes. (please see attachments below) Can you check it out on your end as well?

![image](https://user-images.githubusercontent.com/46043276/103170050-3f4baa00-487c-11eb-87fc-03917a224e99.png)
![image](https://user-images.githubusercontent.com/46043276/103170054-4672b800-487c-11eb-9389-fcbea7c6d0c2.png)

Kurotwo commented on components/RegisterModal.js:
```diff
@@ -0,0 +1,227 @@
+        </Form.Item>
+
+        <Tooltip
+          title="Password must be atleast 6 characters and alphanumeric!"
```
Typo

Kurotwo reviewed (commented):
Hi @jasonevangelista. We appreciate the effort you put in the project! I'd like to comment on some UI and things I saw.  

1. Password is considered valid despite being only alphabetic.
![image](https://user-images.githubusercontent.com/21964136/103170424-17aa1100-487f-11eb-91ae-859e1e17a158.png)

2.  The text for Login and Sign Up in restaurant pages doesn't seem to be vertically centered. I might also suggest making the Login button text white(?) so it contrasts more with the navbar color.
![image](https://user-images.githubusercontent.com/21964136/103170464-622b8d80-487f-11eb-89eb-167a18f27442.png)

3. Typo with the password requirements notif box. 
![image](https://user-images.githubusercontent.com/21964136/103170511-b0409100-487f-11eb-9f6d-3c16c40a61ef.png)

Aside from that, the modal looks great!

jasonevangelista committed:
renamed handleCancel to closeModal

jasonevangelista committed:
adjusted body style to keep scrollbar when modal opens

jasonevangelista committed:
adjusted password alphanumeric validation and notif text

jasonevangelista committed:
adjusted header alignments

jasonevangelista commented:
Thank you for the suggestions and comments!

@NeilMatthewLua 

- opening modal in landing page now still shows the scroll bar

![Screen Shot 2020-12-28 at 3 52 46 PM](https://user-images.githubusercontent.com/50069439/103199136-20edb900-4925-11eb-949f-6fa345663e25.png)

@Kurotwo 

- Adjusted header text alignment and colors

![Screen Shot 2020-12-28 at 3 52 58 PM](https://user-images.githubusercontent.com/50069439/103199221-52668480-4925-11eb-8497-ad6ccbfc0217.png)

- fixed typo in notif box
- password alphanumeric verification now working properly

![Screen Shot 2020-12-28 at 3 53 20 PM](https://user-images.githubusercontent.com/50069439/103199254-67431800-4925-11eb-83b8-f2a2da0644da.png)

NeilMatthewLua commented:
@jasonevangelista 

Maybe we can update the hint to tell users na it wont take special char

![image](https://user-images.githubusercontent.com/46043276/103200783-3ebd1d00-4929-11eb-8244-e1277f7a5f28.png)

Aside from that, the new updates look great!

jasonevangelista commented:
> @jasonevangelista
> 
> Maybe we can update the hint to tell users na it wont take special char
> 
> ![image](https://user-images.githubusercontent.com/46043276/103200783-3ebd1d00-4929-11eb-8244-e1277f7a5f28.png)
> 
> Aside from that, the new updates look great!

hmm something like "Password must be at least 6 characters and must consist of both letters and numbers only" ?

hb yung error message, goods na ba yun na "Password must be alphanumeric!" or change to "Password must  consist of both letters and numbers only"

jasonevangelista commented:
@NeilMatthewLua hb like this? tho havent pushed this sa repo pa HAHAHA
![Screen Shot 2020-12-29 at 3 26 55 PM](https://user-images.githubusercontent.com/50069439/103266345-58726900-49ea-11eb-9907-35e30aacbee3.png)

NeilMatthewLua commented:
@jasonevangelista im so sorry your previous comment didn't notify me HUHU tag me nalang if ever HAHAH.  How about "must only be alphanumeric" yung dulo para less lengthy?

jasonevangelista committed:
adjusted password notif info

jasonevangelista commented:
@NeilMatthewLua no problem!! HAHAHA like this?

![Screen Shot 2020-12-29 at 4 54 05 PM](https://user-images.githubusercontent.com/50069439/103271929-82318d00-49f6-11eb-9cb5-19e345ebfc96.png)

NeilMatthewLua commented:
@jasonevangelista looks good, thanks!!
